src/lib/core/components/Switch.tsx

```diff
@@ -0,0 +1,47 @@
+import { grey, blue } from '@material-ui/core/colors';
+import { FilledSwitch } from '@veupathdb/coreui';
+import { SwitchStyleSpecSubset } from '@veupathdb/coreui/dist/components/widgets/switch';
+
+interface SwitchProps {
+  selectedOption: boolean;
+  disabled?: boolean;
+  onOptionChange?: (selection: boolean) => void;
+  labels?: {
+    left?: string;
+    right?: string;
+  };
+  size?: 'small' | 'medium';
+  styleOverrides?: SwitchStyleSpecSubset;
+}
+
+export const Switch = ({
+  selectedOption,
+  disabled,
+  onOptionChange,
+  labels,
+  size = 'medium',
+  styleOverrides,
+}: SwitchProps) => {
+  return (
+    <FilledSwitch
```

Comment:
Could you help me understand why we'd want to create a `Switch` component in `web-eda` instead of just directly importing `FilledSwitch` from `CoreUI`? I understood why we'd refactor the existing `web-component`'s toggle to use the `CoreUI` switch as its internals. Though we may in the future want to hunt down and replace all imports of the `web-components` switch to instead come from` CoreUI`, it's a simpler and quicker approach to just update the existing component. But I'm not quite seeing a benefit to `web-eda` also having a `Switch` component.

Reply:
Yeah! It's just because we use this same styling of the switch three different places in the EDA, and I thought duplicating the styling would not be very DRY:

```
      options={[false, true]}
      styleOverrides={{
        default: [
          { backgroundColor: grey[500] },
          { backgroundColor: blue[800], knobColor: 'white' },
        ],
        hover: [
          { backgroundColor: grey[500] },
          { backgroundColor: blue[800], knobColor: 'white' },
        ],
        disabled: { backgroundColor: grey[300] },
      }}
```

Reply:
I'm not sure, but I think we should use the same styling everywhere. If we really want different styles in some places, maybe we should make variants in coreui, similar to what was done with [buttons](https://github.com/VEuPathDB/CoreUI/tree/main/src/components/buttons)?

Reply:
Can the switch just use theme colors? For example, blue[800] doesn't match clinepi's teal or mbio's mutedBlue themes. I don't think there was originally a reason that it needed to be gray. 
If we want it to be colored when on and gray when off, could we use the default 'disabled' styling for the component?